dvc/remote/base.py

```diff
@@ -174,13 +381,18 @@ def save(self, path_info, checksum_info):
 
         checksum = checksum_info[self.PARAM_CHECKSUM]
         if not self.changed_cache(checksum):
+            self._checkout(path_info, checksum)
             return
 
-        to_info = self.checksum_to_path_info(checksum)
+        self._save(path_info, checksum)
```

Comment:
This kind of abstraction doesn't look like the best one to me: `foo` calling `_foo`.
If I understand correctly, the reason behind doing this is to have a private method that instances can overwrite to modify the behavior without modifying the public interface.

A good refactor could be to migrate to `ABC` and use `super()` to call the parent's method: https://www.python-course.eu/python3_abstract_classes.php

If this isn't the case and I'm missing something, maybe we should improve the naming, or it is the wrong abstraction, having `save` and `_save` is not intuitive.